pydantic/_internal/_decorators.py

```diff
@@ -56,13 +56,17 @@ class FieldValidatorDecoratorInfo:
         fields: A tuple of field names the validator should be called on.
         mode: The proposed validator mode.
         check_fields: Whether to check that the fields actually exist on the model.
+        input_type: The input type of the function. This is only used to generate
+            the appropriate JSON Schema (in validation mode) and can only specified
+            when `mode` is either `'before'`, `'plain'` or `'wrap'`.
```

Comment:
If this only affects the JSON schema and not any actual validation, I think the name of this argument should reflect that. Otherwise I think you might expect/have a hard time remembering if it validates the value into an int first.

I understand the point of this is to handle non-after modes, so it makes sense for it to not add any validation, but I think you have to think about it a bit hard to remember this. If the argument was just `json_schema_input_type` it's very clear what it means.

Reply:
In all fairness, this is a good argument against my idea to add support for this to the core schema. We can chat about this in more detail on our next catchup 👍